util/src/main/java/org/killbill/billing/util/bcd/BillCycleDayCalculator.java

```diff
@@ -76,22 +76,27 @@ public static LocalDate alignProposedBillCycleDate(final DateTime proposedDate,
     }
 
 
-    public static LocalDate alignProposedNextBillCycleDate(final LocalDate proposedDate, final int billingCycleDay, final BillingPeriod billingPeriod) {
+    public static LocalDate alignProposedNextBillCycleDate(final LocalDate proposedDate, final LocalDate originalDate, final int billingCycleDay, final BillingPeriod billingPeriod) {
```

Comment:
Same thing here - rename and swap:
* `originalDate` -> `prevTransitionDate`
* `proposedDate` -> `curTransitionDate`
*

Reply:
What about we also change the method name to `alignProposedNextBillCycleDate` -> `alignToNextBillCycleDate`. It would take as a first argument `prevTransitionDate`, and then `curTransitionDate` with a comment /* date to be aligned */  ?